spec/features/admin/orders_spec.rb

```diff
@@ -283,6 +283,19 @@ def new_order_with_distribution(distributor, order_cycle)
         end
       end
 
+      scenario "shows only eligible adjustments" do
+        adjustment = create(
+          :adjustment,
+          adjustable: @order,
+          label: "invalid adjustment",
+          amount: 0
+        )
+
+        visit spree.edit_admin_order_path(@order)
+
+        expect(page).to have_no_content adjustment.label
+      end
```

Comment:
Have you considered writing a controller test instead? the test does not involve any user interaction or more than a page so it should be possible AFAIK. You know the benefits already, improved reliability, execution speed, etc.

Reply:
@sauloperez the filtering is done in the view

Reply:
I see, and that is why I suggest a controller test. By using `render_views`, which we might use by default, we hit the view too.

Reply:
eheh, good that I commented, I learned something. I didnt know that render_views trick 👍

Reply:
Good idea. Have a look at my last commit. We are saving 7 seconds here.

Reply:
Wow! 7 seconds it's a whole lot!